### PR TITLE
Load Salt Cloud Hetzner Module Config from Usual Sources

### LINKED ISSUE #61399: [BUG] Hetzner Cloud state ignores config values
**Description**
Using the [Hetzner Cloud module](https://docs.saltproject.io/en/master/ref/clouds/all/salt.cloud.clouds.hetzner.html) in a state ignores the configured values in `/etc/salt/cloud.providers.d/` or `/etc/salt/cloud.profiles.d/` and only takes state attributes, as mentioned in #61392.

**Setup**

I installed `salt-minion`, `salt-master` and `salt-cloud` on a Hetzner Cloud Debian 11 machine, generated said ssh key, added it to the hetzner cloud, generated an API key there and added some configs to my machine:

`/etc/salt/cloud.providers.d/hetzner.conf`:

```yaml
hetzner:
  key: <my API key>
  private_key: /root/.ssh/id_ed25519
  ssh_keys:
    - id_ed25519
  location: fsn1
  driver: hetzner
  minion:
    <additional minion conf>
```

`/etc/salt/cloud.profiles.d/hetzner.conf`:

```yaml
cx21_hetzner:
  provider: hetzner
  image: debian-11
  size: cx21
  location: nbg1
```

My `cloud-hetzner.sls`:

```yaml
my-instance-name:
  cloud.present:
    - name: <machine name>
    - cloud_provider: hetzner
    - image: debian-11
    - size: cx21
 ```

**Steps to Reproduce the behavior**

1. Apply the mentioned state, the new VM will ignore `location` and `ssh_keys`, if not configured in the state.

**Expected behavior**

Search and return a setting in a known order:

1. In the virtual machine's configuration
2. In the virtual machine's profile configuration
3. In the virtual machine's provider configuration
4. In the salt cloud configuration if global searching is enabled
5. Return the provided default

**Versions Report**

<details><summary>salt --versions-report</summary>

```yaml
Salt Version:
          Salt: 3004
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.3
       libgit2: Not Installed
      M2Crypto: 0.37.1
          Mako: Not Installed
       msgpack: 1.0.0
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.9.7
        pygit2: Not Installed
        Python: 3.9.2 (default, Feb 28 2021, 17:03:44)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 20.0.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4
 
System Versions:
          dist: debian 11 bullseye
        locale: utf-8
       machine: x86_64
       release: 5.10.0-9-amd64
        system: Linux
       version: Debian GNU/Linux 11 bullseye
```

</details>

**Additional context**

Using `config.get_cloud_config_value()` instead of `vm_.get()` provides the expected behaviour. PR incoming.

### PR DESCRIPTION
**Pull request by HerHde**
### What does this PR do?
Fix config loading behaviour

### What issues does this PR fix or reference?
Fixes: #61399

### Previous Behavior
Config values were only taken from the current context, as available in `vm_`

### New Behavior
Search and return a setting in a known order:

1. In the virtual machine's configuration
2. In the virtual machine's profile configuration
3. In the virtual machine's provider configuration
4. In the salt cloud configuration if global searching is enabled
5. Return the provided default

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?

Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by HerHde** [comment]
I don't understand why the bot still mentions missing docstrings, did I not fix them?

**Comment by HerHde** [comment]
Hey @dwoz, this is ready for review.

As mentioned, I don't quite understand why the PR Bot is complaining. Some pipelines seem to fail while trying to check out the code, so I assume it shouldn't concern me.

If there is anything, please let me know!

**Comment by HerHde** [comment]
Maybe @Ch3LL could help me with this one?

File: salt/cloud/clouds/hetzner.py [hunk 1]
```diff
@@ -289,7 +329,10 @@ def create(vm_):
     )
 
     # Get the ssh_keys
-    ssh_keys = vm_.get("ssh_keys", None)
+    ssh_keys = config.get_cloud_config_value(
+        "ssh_keys", vm_, __opts__, search_global=False, default=False
```
Comment (Ch3LL) [inline salt/cloud/clouds/hetzner.py:333]: Why are you setting the default to be False here? It looks like it should stay set to None.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/cloud/clouds/hetzner.py (hunk 1 above)
Comment (HerHde) [reply salt/cloud/clouds/hetzner.py:333]: Good question, I think I got inspired by [the digitalocean driver](https://github.com/saltstack/salt/blob/master/salt/cloud/clouds/digitalocean.py#L323) here, but `None` seems adequate, indeed.

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**